frontend/src/components/activePage/TableWorkspaceLogic/TableWorkspaceLogic.tsx

```diff
@@ -36,179 +33,93 @@ export interface ITableWorkspaceLogicProps {
+    ({ id }) => id
+  )})`;
+
   const {
     loading: loadingInstances,
     error: errorInstances,
     subscribeToMore: subscribeToMoreInstances,
   } = useInstancesLabelSelectorQuery({
     variables: {
-      labels: `crownlabs.polito.it/workspace in (${workspaces.map(
-        ({ id }) => id
-      )})`,
+      labels: label,
     },
     onCompleted: setDataInstances,
     fetchPolicy: fetchPolicy_networkOnly,
   });
 
   useEffect(() => {
-    subscribeToMoreInstances({
-      document: UpdatedInstancesLabelSelectorDocument,
-      variables: {
-        tenantNamespace,
-      },
-      updateQuery: (prev, { subscriptionData }) => {
-        const {
-          data,
-        } = subscriptionData as UpdatedInstancesLabelSelectorSubscriptionResult;
-
-        if (!data?.updateInstanceLabelSelector?.instance) return prev;
-
-        const { instance, updateType } = data?.updateInstanceLabelSelector;
-
-        if (prev.instanceList?.instances) {
-          let instances = prev.instanceList.instances;
-          if (updateType === UpdateType.Deleted) {
-            instances = instances.filter(i => {
-              if (i?.metadata?.name !== instance.metadata?.name) {
-                return true;
-              } else {
-                notification.warning({
-                  message:
-                    i?.spec?.templateCrownlabsPolitoItTemplateRef
-                      ?.templateWrapper?.itPolitoCrownlabsV1alpha2Template?.spec
-                      ?.templateName,
-                  description: `${instance.metadata?.name} deleted`,
-                });
-                return false;
-              }
-            });
-          } else {
-            if (
-              instances.find(i => i?.metadata?.name === instance.metadata?.name)
-            ) {
-              instances = instances.map(i =>
-                i?.metadata?.name === instance.metadata?.name ? instance : i
+    if (!loadingInstances) {
+      subscribeToMoreInstances({
+        document: UpdatedInstancesLabelSelectorDocument,
+        variables: {
+          tenantNamespace,
+        },
+        updateQuery: (prev, { subscriptionData }) => {
+          const {
+            data,
+          } = subscriptionData as UpdatedInstancesLabelSelectorSubscriptionResult;
+
+          if (!data?.updateInstanceLabelSelector?.instance) return prev;
+
+          const { instance, updateType } = data?.updateInstanceLabelSelector;
+
+          if (prev.instanceList?.instances) {
+            let instances = [...prev.instanceList.instances];
+            if (updateType === UpdateType.Deleted) {
+              instances = instances.filter(
+                i => i?.metadata?.name !== instance.metadata?.name
               );
             } else {
-              instances = [...instances, instance];
+              if (
+                instances.find(
+                  i => i?.metadata?.name === instance.metadata?.name
+                )
```

Comment:
```suggestion
                instances.find(matchInstance(instance))
```
I'd make a helper function like this, to be put in the utils:
```ts
export function matchInstance(instance: Instance) : (Instance) => bool {
  return (i: instance) => i?.metadata?.name === instance.metadata?.name
}
```
Consider making something equivalent for the map.
These would be used in several places it seems...
(p.s. I'm not really sure about the syntax, but it's for giving you an idea)

Reply:
in this cases it's a little bit a problem, because the parameters that have to be passed are not 'Instance' type, but something that comes from the subscription and the type is not too much defined

Reply:
well, you still know they both have metadata and name, you can make a simple interface for them both